src/verify.cc

```diff
@@ -152,6 +153,38 @@ bool verifySignatureOct(absl::string_view key, const EVP_MD* md,
                             castToUChar(signed_data), signed_data.length());
 }
 
+bool verifySignatureEd25519(EVP_PKEY* key, const uint8_t* signature,
+                            size_t signature_len, const uint8_t* signed_data,
+                            size_t signed_data_len) {
+  if (key == nullptr || signature == nullptr || signed_data == nullptr) {
+    return false;
+  }
+
+  if (signature_len != ED25519_SIGNATURE_LEN) {
+    return false;
+  }
+
+  uint8_t raw_key[ED25519_PUBLIC_KEY_LEN];
+  size_t out_len = ED25519_PUBLIC_KEY_LEN;
+  if (EVP_PKEY_get_raw_public_key(key, raw_key, &out_len) != 1 ||
```

Comment:
The other option would have been to store the raw key in the Pubkey struct, rather than the key wrapped in an EVP_PKEY object. However I couldn't come up with the right C++ syntax (presumably using a unique_ptr around maybe a std::array of uint8_t data?) to make it work. Storing the key as an EVP_PKEY fits in better with the other code, at the expense of needed to extract the raw key here on every token validation. It seems to be an oddity of the ed25519 implementation in openssl/boringssl that raw keys are used instead of some sort of key object like EVP_PKEY, EC_KEY, etc.

Reply:
we could just use std::string in `struct Pubkey` to store this raw key

std::string okp_raw_key_;

okp_raw_key_.resize(ED25519_PUBLIC_KEY_LEN);
use its buffer as okp_raw_key_.data(), 

For better performance, its better to get the raw key during jwks parsing.